libbeat/outputs/elasticsearch/output.go

```diff
@@ -123,17 +121,14 @@ func (out *elasticsearchOutput) init(
 
 // loadTemplate checks if the index mapping template should be loaded
 // In case template loading is enabled, template is written to index
-func loadTemplate(config Template, clients []mode.ProtocolClient) {
+func loadTemplate(config Template, client *Client) {
```

Comment:
Yeah, I think it's safest to do this for each host, because inserting even one document before the template is loaded makes the whole effort useless.

I don't think the races can cause problems because we're loading the same template, but adding a mutex shouldn't hurt either.